### PR TITLE
Do not force resolution to IPV4 addresses

### PR DESCRIPTION
**Pull request by pakrym-stripe**
Stripe does not have an AAAA record so it's impossible to resolve an IPv6 address for `api.stripe.com`.

Forcing resolution to IPv4 via `CURL_IPRESOLVE_V4` [interferes with IPv4 over IPv6 proxy network setups and DNS64](https://github.com/stripe/stripe-php/pull/1046#issuecomment-1886833339).

We also don't enforce the same setting in any of the other libraries. 

## Changelog

* Do not force resolution to IPv4 - Forcing IPv4 was causing hard-to-understand failures for users in IPv6-only environments. If you want to force IPv4 yourself, you can do so by telling the API client to use a CurlClient other than the default
```
$curl = new \Stripe\HttpClient\CurlClient([CURLOPT_IPRESOLVE => CURL_IPRESOLVE_V4]);
\Stripe\ApiRequestor::setHttpClient($curl);
```

### CODE REVIEW

**Review by richardm-stripe — APPROVED** [review]
LGTM. Let's write a careful `## Changelog` in the PR description so that if any users begin suffering from #1045, they can find the workaround and we can investigate.

**Comment by pakrym-stripe** [comment]
We'll wait until the next major to merge this.

**Comment by ramya-stripe** [comment]
@pakrym-stripe, @richardm-stripe, Reviving this PR with the base as the next-major branch. This is stiil a go?

**Comment by pakrym-stripe** [comment]
@ramya-stripe , yes, but needs a changelog entry as @richardm-stripe mentioned.

**Pull request merged**
